pheget/static/js/searchbox.js

```diff
@@ -1,60 +1,113 @@
-        }
-    };
-    xhr.send();
-};
+// Generic failure callback function
+function failureCallback(error) {
+    console.log('Error status: ' + error);
+}
 
-// eslint-disable-next-line no-unused-vars
-function parseSearch(searchText) {
+
+// Returns the data from an omnisearch fetch if successful, return nothing otherwise
+function getOmniSearch(searchText) {
+    const omniurl = `https://portaldev.sph.umich.edu/api/v1/annotation/omnisearch/?q=${searchText}&build=GRCh38`;
+    var data = fetch(omniurl)
+        .then((response) => response.json())
+        .then((myJson) => myJson.data[0])
+        .catch(function(error) {
+            console.log('Omnisearch error, status code: ' + error);
+            return;
+        });
+    return data;
+}
+
+
+// Returns the data from our internal best variant API by searching for the gene_id (ENSG...) or gene_name
+function getBestVar(symbol) {
+    const bestVarURL = `/api/bestvar/${symbol}`;
+    var data = fetch(bestVarURL)
+        .then((response) => response.json())
+        .then((resp) => resp.data)
+        .catch(failureCallback);
+    return data;
+}
+
+
+// Define a function that returns the search query type to the parseSearch function
+// We expect queries in the form of a single variant (chr:pos or rsnum), a range (chr:start-end), or a gene name (ENSG or symbol)
+function parseSearchText(searchText) {
+    // Use regular expressions to match known patterns for single variant, range, and rs number
     const chromposPattern = /(chr)?([1-9][0-9]?|X|Y|MT):([1-9][0-9]+)/;
     const rangePattern = /(chr)?([1-9][0-9]?|X|Y|MT):([1-9][0-9]+)-([1-9][0-9]+)/;
+    const rsPattern = /(rs[1-9][0-9]+)/;
     const cMatch = searchText.match(chromposPattern);
     const cMatchRange = searchText.match(rangePattern);
-    // If input is in 'chrom:pos' format (with or without 'chr'), go directly to the single-variant page
+    const cMatchRS = searchText.match(rsPattern);
+    // If input is in 'chrom:pos' format (with or without 'chr'), return the position of the single variant
     // TODO: Show closest variant if specific variant does not exist
     if (cMatch !== null && searchText === cMatch[0]) {
         const chrom = cMatch[2];
         const pos = cMatch[3];
-        window.location = `/variant/${chrom}_${pos}`;
+        return {'type': 'variant', 'chrom': `${chrom}`, 'start': `${pos}`, 'end': `${pos}`};
     } else if (cMatchRange !== null && searchText === cMatchRange[0]) {
-        // If the query is a range in the form [chr]:[start]-[end], go to a region view page of that range
+        // If the query is a range in the form [chr]:[start]-[end], return the position of the range
         const chrom = cMatchRange[2];
         const start = cMatchRange[3];
         const end = cMatchRange[4];
-        window.location = `/region/?chrom=${chrom}&start=${start}&end=${end}`;
+        return {type: 'range', chrom: `${chrom}`, start: `${start}`, end: `${end}`};
```

Comment:
Do we need the string template literals here? (`chrom: ${chrom}`)

Cute syntax trick: if the variable is the same name as an object key, then these two are equivalent:

`{chrom: chrom}` vs `{ chrom }`

Reply:
Thanks for the note on syntax, these have been fixed and changed to Promises. All returns from parseSearchText are promises, and should be handled the same way by parseSearch.